admin/helpers/check-window.js

```diff
@@ -0,0 +1,110 @@
+    adminEndDay: checkWindowData.adminEndDate.format('D'),
+    adminEndMonth: checkWindowData.adminEndDate.format('MM'),
+    adminEndYear: checkWindowData.adminEndDate.format('YYYY'),
+    adminEndDateDisabled: currentDate.isSameOrAfter(checkWindowData.adminEndDate),
+    checkWindowUrlSlug: checkWindowData.urlSlug,
+    checkWindowName: checkWindowData.name,
+    familiarisationCheckStartDate: checkWindowData.familiarisationCheckStartDate,
+    familiarisationCheckStartDay: checkWindowData.familiarisationCheckStartDate.format('D'),
+    familiarisationCheckStartMonth: checkWindowData.familiarisationCheckStartDate.format('MM'),
+    familiarisationCheckStartYear: checkWindowData.familiarisationCheckStartDate.format('YYYY'),
+    familiarisationCheckStartDateDisabled: currentDate.isSameOrAfter(checkWindowData.familiarisationCheckStartDate),
+    familiarisationCheckEndDate: checkWindowData.familiarisationCheckEndDate,
+    familiarisationCheckEndDay: checkWindowData.familiarisationCheckEndDate.format('D'),
+    familiarisationCheckEndMonth: checkWindowData.familiarisationCheckEndDate.format('MM'),
+    familiarisationCheckEndYear: checkWindowData.familiarisationCheckEndDate.format('YYYY'),
+    familiarisationCheckEndDateDisabled: currentDate.isSameOrAfter(checkWindowData.familiarisationCheckEndDate),
+    liveCheckStartDate: checkWindowData.checkStartDate,
+    liveCheckStartDay: checkWindowData.checkStartDate.format('D'),
+    liveCheckStartMonth: checkWindowData.checkStartDate.format('MM'),
+    liveCheckStartYear: checkWindowData.checkStartDate.format('YYYY'),
+    liveCheckStartDateDisabled: currentDate.isSameOrAfter(checkWindowData.checkStartDate),
+    liveCheckEndDate: checkWindowData.checkEndDate,
+    liveCheckEndDay: checkWindowData.checkEndDate.format('D'),
+    liveCheckEndMonth: checkWindowData.checkEndDate.format('MM'),
+    liveCheckEndYear: checkWindowData.checkEndDate.format('YYYY'),
+    liveCheckEndDateDisabled: currentDate.isSameOrAfter(checkWindowData.checkEndDate),
+  }
+  checkWindowViewData.adminPeriodDisabled = checkWindowViewData.adminStartDateDisabled && checkWindowViewData.adminEndDateDisabled
+  checkWindowViewData.familiarisationCheckPeriodDisabled = checkWindowViewData.familiarisationCheckStartDateDisabled && checkWindowViewData.familiarisationCheckEndDateDisabled
+  checkWindowViewData.liveCheckPeriodDisabled = checkWindowViewData.liveCheckStartDateDisabled && checkWindowViewData.liveCheckEndDateDisabled
+  checkWindowViewData.pastCheckWindow = checkWindowViewData.adminPeriodDisabled && checkWindowViewData.familiarisationCheckPeriodDisabled && checkWindowViewData.liveCheckPeriodDisabled
+  return checkWindowViewData
+}
+
+/**
+ * Prepare validated data for submission
+ * @param {Object} requestData
+ * @param {Number} checkWindowId
+ * @returns {Object} Formatted check window data
+ */
+checkWindowHelper.prepareSubmissionData = (requestData, checkWindowId = null) => {
```

Comment:
Is this business logic?  There is a hard-coded check end date at the bottom of the function.

Reply:
@jon-shipley The familiarisation and live check end dates are actually required to be set at the end time of that dates specified by the user. However you got he thinking about relocating this method elsewhere since its more of business logic.

Reply:
@GeorgeChatzigiannis if this is structuring the raw data service output that would be useless to anyone without the structuring, then you may want to consider this as business logic that belongs in the service.  However, if this is a formatting exercise just for the web view, put it in a presenter.

Reply:
@GuyHarwood I moved this to the service level since it is indeed business logic.